atlp-rwanda/Dominators-Phantom-BE
# Ft enquire buses info

### What does this PR do?
- changed a `currentLocation` in `simulateController.js` to have a `fromCoordinates`
- Adds a route for getting all info about a particular bus
### Description of Task to be completed?

### How should this be manually tested?
- After cloning the repo run `npm i` to install all packages
- For Windows users refer an extra step for installing [`redis`](https://www.youtube.com/watch?v=188Fy-oCw4w&ab_channel=ProgrammingKnowledge2)
- use postman to test different endpoints present in [`index.js`](https://github.com/atlp-rwanda/Dominators-Phantom-BE/blob/develop/src/routes/index.js)
- Run `npm run migrate` to migrate all models in the local database
- Run `npm run seed` to populate the tables with seeders' values
- Run `npm run dev` to start the server manually
- use `http://localhost:3000/api/v1/login` to log in by using admin credentials present in the seeder of the user table. (Change the port **3000** to adhere to your chosen port)
- After login and adding Authorisation in headers, insert the URL with the port and the method e.g., `http://localhost:3000/api/v1/bus-info/:busId` and `GET`
- The expected response for a `"busId": "a097902c-0e03-493c-b010-6556e9dc6d77"`  which has been assigned a driver of  `"id": "7ec25dc7-ccbe-4101-b535-5007abf49600"`, and which undergoes a journey should look like this
```
{
    "data": {
        "busId": "a097902c-0e03-493c-b010-6556e9dc6d77",
        "prateNumber": "RAB309",
        "busType": "YUTONGO",
        "allDriverBus": [
            {
                "id": "7ec25dc7-ccbe-4101-b535-5007abf49600",
                "firstName": "Jeie",
                "lastName": "Umusi",
                "email": "j.umuhire@alustudent.com",
                "role": "driver"
            }
        ],
        "busRoute": {
            "routeId": "b5e2cedc-0c70-4449-a25f-1c42149f3162",
            "route": "Kigali to Huye",
            "distance": "200km",
            "status": "pending",
            "fromCoordinates": [
                "1.234567",
                "32.03456"
            ],
            "toCoordinates": [
                "10.234567",
                "32.05556"
            ]
        },
        "journey": [
            {
                "journeyId": "278fbec5-cf52-4b57-9e37-8527549df5d2",
                "journeyTitle": "Kigali - Huye Journey",
                "currentLocation": [
                    "1.234567",
                    "32.03456"
                ],
                "speed": 60,
                "passengers": 56,
                "routeID": "b5e2cedc-0c70-4449-a25f-1c42149f3162",
                "driverID": "7ec25dc7-ccbe-4101-b535-5007abf49600",
                "busID": "a097902c-0e03-493c-b010-6556e9dc6d77",
                "status": "Moving",
                "trafficStatus": "None",
                "createdAt": "2022-06-30T16:49:14.423Z",
                "updatedAt": "2022-06-30T16:49:18.506Z"
            }
        ]
    }
}
```
- Otherwise, after running seed the `api/v1/bus-info/:busId` will give a response for the bus in seeders which looks like the following
```
{
    "data": {
        "busId": "65106635-368c-4364-9432-249b00643427",
        "prateNumber": "RAB309",
        "busType": "YUTONGO",
        "allDriverBus": [],
        "busRoute": {
            "routeId": "4a38496d-19a4-408d-a513-4353ba075cc1",
            "route": "Kigali to Huye",
            "distance": "200km",
            "status": "pending",
            "fromCoordinates": [
                "1.234567",
                "32.03456"
            ],
            "toCoordinates": [
                "10.234567",
                "32.05556"
            ]
        },
        "journey": []
    }
}
```

### Any background context you want to provide?
The old seeder of the bus was deleted from this branch to avoid conflicts while migrating and seeding the tables. It is recommended that the one in `develop` called `20220603114223-BusSeed.js` can be deleted too. 
### What are the relevant pivotal tracker stories?

### Screenshots (if appropriate)

### Questions:
